File: src/api/LMSEntities/DataTransferObjects/LibraryCardForUpdate.cs
using System.ComponentModel.DataAnnotations;

namespace LMSEntities.DataTransferObjects
{
    public class LibraryCardForUpdate
    {
        public int Id { get; set; }
        [Required]
        [StringLength(25, MinimumLength = 2)]
        public string FirstName { get; set; }
        [Required]
        [StringLength(25, MinimumLength = 2)]
        public string LastName { get; set; }
        [Required]
        [EmailAddress]
        public string Email { get; set; }
        [Required]
        [Phone]
        public string PhoneNumber { get; set; }
        public AddressDto Address { get; set; }
        // public int AddressId { get; set; }
        public string Gender { get; set; }
        public string Status { get; set; }
    }
}

File: src/spa/src/dto/models/library-card-for-update.ts
/* tslint:disable */
/* eslint-disable */
import { AddressDto } from './address-dto';
import { MemberGenderDto } from './member-gender-dto';
export interface LibraryCardForUpdate {
  address: AddressDto;
  email: string;
  firstName: string;
  gender: MemberGenderDto;
  id: number;
  lastName: string;
  phoneNumber: string;
}
